hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/impl/prefetch/BlockManagerParams.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hadoop.fs.impl.prefetch;
+
+import org.apache.hadoop.classification.InterfaceAudience;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.LocalDirAllocator;
+import org.apache.hadoop.fs.statistics.DurationTrackerFactory;
+
+/**
+ * This class is used to provide params to {@link BlockManager}.
+ */
+@InterfaceAudience.Private
+public final class BlockManagerParams {
+
+  /**
+   * Asynchronous tasks are performed in this pool.
+   */
+  private final ExecutorServiceFuturePool futurePool;
+  /**
+   * Information about each block of the underlying file.
+   */
+  private final BlockData blockData;
+  /**
+   * Size of the in-memory cache in terms of number of blocks.
+   */
+  private final int bufferPoolSize;
+  /**
+   * Statistics for the stream.
+   */
+  private final PrefetchingStatistics prefetchingStatistics;
+  /**
+   * The configuration object.
+   */
+  private final Configuration conf;
+  /**
+   * The local dir allocator instance.
+   */
+  private final LocalDirAllocator localDirAllocator;
+  /**
+   * Max blocks count to be kept in cache at any time.
+   */
+  private final int maxBlocksCount;
+  /**
+   * Tracker with statistics to update.
+   */
+  private final DurationTrackerFactory trackerFactory;
+
+  @SuppressWarnings("checkstyle:parameternumber")
+  private BlockManagerParams(ExecutorServiceFuturePool futurePool, BlockData blockData,
```

Comment:
we're still ending up with a constructor here. Instead of this, look at [S3ClientFactory.S3ClientCreationParameters](https://github.com/apache/hadoop/blob/trunk/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ClientFactory.java#L98) to see how we get and set parameters there, we should follow the same patter here.

Reply:
done